plugins/woocommerce/src/Internal/Admin/ProductReviews/ReviewsListTable.php

```diff
@@ -0,0 +1,1490 @@
+		$product = $product_id ? wc_get_product( $product_id ) : null;
+
+		if ( $product instanceof WC_Product ) {
+			$this->current_product_for_reviews = $product;
+		}
+	}
+
+	/**
+	 * Sets the `$comment_status` global based on the current request.
+	 *
+	 * @global string $comment_status
+	 *
+	 * @return void
+	 */
+	protected function set_review_status() : void {
+		global $comment_status;
+
+		$comment_status = sanitize_text_field( wp_unslash( $_REQUEST['comment_status'] ?? 'all' ) ); // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+
+		if ( ! in_array( $comment_status, [ 'all', 'moderated', 'approved', 'spam', 'trash' ], true ) ) {
+			$comment_status = 'all'; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+		}
+	}
+
+	/**
+	 * Sets the `$comment_type` global based on the current request.
+	 *
+	 * @global string $comment_type
+	 *
+	 * @return void
+	 */
+	protected function set_review_type() : void {
+		global $comment_type;
+
+		$review_type = sanitize_text_field( wp_unslash( $_REQUEST['review_type'] ?? 'all' ) );
+
+		if ( 'all' !== $review_type && ! empty( $review_type ) ) {
+			$comment_type = $review_type; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+		}
+	}
+
+	/**
+	 * Builds the `orderby` and `order` arguments based on the current request.
+	 *
+	 * @return array
+	 */
+	protected function get_sort_arguments() : array {
+		$orderby = sanitize_text_field( wp_unslash( $_REQUEST['orderby'] ?? '' ) );
+		$order   = sanitize_text_field( wp_unslash( $_REQUEST['order'] ?? '' ) );
+
+		$args = [];
+
+		if ( ! in_array( $orderby, $this->get_sortable_columns(), true ) ) {
+			$orderby = 'comment_date_gmt';
+		}
+
+		// If ordering by "rating", then we need to adjust to sort by meta value.
+		if ( 'rating' === $orderby ) {
+			$orderby          = 'meta_value_num';
+			$args['meta_key'] = 'rating';
+		}
+
+		if ( ! in_array( strtolower( $order ), [ 'asc', 'desc' ], true ) ) {
+			$order = 'desc';
+		}
+
+		return wp_parse_args(
+			[
+				'orderby' => $orderby,
+				'order'   => strtolower( $order ),
+			],
+			$args
+		);
+	}
+
+	/**
+	 * Builds the `type` argument based on the current request.
+	 *
+	 * @return array
+	 */
+	protected function get_filter_type_arguments() : array {
+
+		$args      = [];
+		$item_type = isset( $_REQUEST['review_type'] ) ? sanitize_text_field( wp_unslash( $_REQUEST['review_type'] ) ) : 'all';
+
+		if ( 'all' === $item_type ) {
+			return $args;
+		}
+
+		$args['type'] = $item_type;
+
+		return $args;
+	}
+
+	/**
+	 * Builds the `meta_query` arguments based on the current request.
+	 *
+	 * @return array
+	 */
+	protected function get_filter_rating_arguments() : array {
+
+		$args = [];
+
+		if ( empty( $this->current_reviews_rating ) ) {
+			return $args;
+		}
+
+		$args['meta_query'] = [
+			[
+				'key'     => 'rating',
+				'value'   => (int) $this->current_reviews_rating,
+				'compare' => '=',
+				'type'    => 'NUMERIC',
+			],
+		];
```

Comment:
A little unsure if this is something to change or not ... but by applying this filter we will stop matching replies (because they lack rating meta data) so on using this filter we suddenly go from something  like:

| | Type | Rating | Review | 
| --- | --- | --- | --- |
| ◽ | Review | ✩ ✩ | Is ok, wrong color though |
| ◽ | Reply |  | Sorry about that |
| ◽ | Review | ✩ ✩ ✩ ✩ | Works well |
| ◽ | Reply |  | Thanks! |

To:

| | Type | Rating | Review | 
| --- | --- | --- | --- |
| ◽ | Review | ✩ ✩ ✩ ✩ | Works well |
| ◽ | Review | ✩ ✩ | Is ok, wrong color though |

(Suddenly a set of rows have vanished) ... it's not wrong, but it also seems slightly jarring [?] ... To draw a parallel, if I was to order the products list table by SKU then those without a SKU do not suddenly disappear.

What are your thoughts? I'm a little on the fence, but wanted to pause and ask.

Reply:
I see what you mean. Could be worth checking what's the current behavior for this on the vanilla comments/reviews page or third party plugins like Product Reviews Pro, to see if there's somehow another pattern that merchants might expect. I wonder if the merchant wants to focus on the reviews and not the replies when using this filter?  Would it be acceptable wait for users feedback on this particular filter to see if it's something we should adjust at all?

Reply:
>  Would it be acceptable wait for users feedback on this particular filter to see if it's something we should adjust at all?

Yes, that's a good course of action :+1: